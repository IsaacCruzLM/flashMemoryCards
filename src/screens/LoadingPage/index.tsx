import React from 'react';
import {View, ActivityIndicator} from 'react-native';

import Theme from '../../styles/themes';
import styles from './styles';

const LoadingPage = () => (
  <View testID="loading-container" style={styles.container}>
    <ActivityIndicator
      testID="loading-indicator"
      size="large"
      color={Theme.colors.primary}
    />
  </View>
);

export default LoadingPage;
