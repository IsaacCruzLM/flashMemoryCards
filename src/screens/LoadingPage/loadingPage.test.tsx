import React from 'react';
import renderer from 'react-test-renderer';
import LoadingPage from './index';

describe('LoadingPage', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<LoadingPage />).toJSON();
    expect(tree).toBeTruthy();
    expect(tree).toMatchSnapshot();
  });

  it('should render with loading indicator', () => {
    const testRenderer = renderer.create(<LoadingPage />);
    const testInstance = testRenderer.root;

    const activityIndicator = testInstance.findByProps({
      testID: 'loading-indicator',
    });

    expect(activityIndicator).toBeTruthy();
  });

  it('should apply container styles', () => {
    const testRenderer = renderer.create(<LoadingPage />);
    const testInstance = testRenderer.root;

    const container = testInstance.findByProps({
      testID: 'loading-container',
    });

    expect(container.props.style).toEqual(
      expect.objectContaining({
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
      }),
    );
  });
});
