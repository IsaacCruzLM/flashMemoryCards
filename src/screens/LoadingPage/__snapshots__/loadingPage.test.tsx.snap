// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingPage should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="loading-container"
>
  <ActivityIndicator
    color="#22272E"
    size="large"
    testID="loading-indicator"
  />
</View>
`;
